remote-api/src/main/java/org/alfresco/rest/api/impl/DownloadsImpl.java

```diff
@@ -87,6 +100,8 @@ public Download createDownloadNode(Download download)
         
         checkNodeIdsReadPermission(zipContentNodeRefs);
         
+        checkArchiveStatus(zipContentNodeRefs);

```

Comment:
Add discovery - only makes sense to call if either S3 Connector (or Azure Connector in future) are deployed.

Reply:
Thanks, I've added the discovery using the `ModuleService` and add a hardcoded limit of 500 to the number of checks.